detect_secrets_server/adhoc/github/webhook.py

```diff
@@ -0,0 +1,90 @@
+"""
+For organizations that integrate with Github, they have the ability to setup
+a webhook to receive events for all repos under the entire organization. In
+such cases, this script allows you to scan other fields in which secrets may
+be leaked, rather than just focusing on secrets in code.
+"""
+import io
+from contextlib import redirect_stdout
+from typing import Any
+from typing import Dict
+from typing import Tuple
+
+from detect_secrets.main import main as run_detect_secrets
+
+
+def scan_for_secrets(event_type: str, body: Dict[str, Any]) -> str:
+    """
+    :param event_type: a full list can be found
+        https://developer.github.com/v3/activity/events/types/
+
+    :returns: link to field with leaked secret
+    """
+    mapping = {
+        'commit_comment': _parse_comment,
+        'issue_comment': _parse_comment,
+        'pull_request_review_comment': _parse_comment,
+        'issues': _parse_issue,
+        'pull_request': _parse_pull_request,
+
+        # NOTE: We're currently ignoring `project*` events, because we don't use
+        #       it. Pull requests welcome!
+    }
+    try:
+        payload, attribution_link = mapping[event_type](body)
+    except KeyError:
+        # Not an applicable event.
+        return None
+
+    f = io.StringIO()
+    with redirect_stdout(f):
+        run_detect_secrets([
+            'scan',
+            '--string', payload,
+        ])
+
+    has_results = any([
+        line
+        for line in f.getvalue().splitlines()
+        if 'True' in line.split(':')[1]
+    ])
+
+    return attribution_link if has_results else None
+
+
+def _parse_comment(body: Dict[str, Any]) -> Tuple[str, str]:
+    if body.get('action', 'created') == 'deleted':
```

Comment:
I could be wrong, but I don't think this does what you want it to.

'created' is [the return value of .get if 'action' is not a present key](https://www.programiz.com/python-programming/methods/dictionary/get).

Reply:
Yeah, that's accurate. Some comments don't have `action` on them, and we want to scan the contents as default behavior.

Reply:
I guess my feedback is since `'created' == 'deleted'` will always be false, it feels weird, but not a strong opinion.